integration_test/tests/power_bi_tests.rs

```diff
@@ -9,11 +9,54 @@ mod integration {
         AttrConnectionPooling, AttrOdbcVersion, ConnectionAttribute, DriverConnectOption,
         EnvironmentAttribute, HDbc, HEnv, Handle, HandleType, InfoType, Pointer, SQLAllocHandle,
         SQLDriverConnectW, SQLFreeHandle, SQLGetInfoW, SQLSetConnectAttrW, SQLSetEnvAttr, SmallInt,
-        SqlReturn,
+        SqlReturn, USmallInt,
     };
     use std::ptr::null_mut;
     use std::slice;
 
+    macro_rules! test_get_info {
+        ($conn_handle:expr, $env_handle: expr, $info_type: expr , $actual_value_modifier: ident) => {{
+            let conn_handle = $conn_handle;
+            let env_handle = $env_handle;
+            let info_type = $info_type;
+            let actual_value_modifier = $actual_value_modifier;
+
+            let str_len_ptr = &mut 0;
+            const BUFFER_LENGTH: SmallInt = 300;
+            let output_buffer = &mut [0u16; (BUFFER_LENGTH as usize - 1)] as *mut _;
+
+            let outcome = SQLGetInfoW(
+                conn_handle as HDbc,
+                info_type,
+                output_buffer as Pointer,
+                BUFFER_LENGTH,
+                str_len_ptr,
+            );
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                outcome,
+                "Expected {}, got {}. Diagnostic message is: {}",
+                sql_return_to_string(SqlReturn::SUCCESS),
+                sql_return_to_string(outcome),
+                get_sql_diagnostics(HandleType::Env, env_handle as Handle)
```

Comment:
```suggestion
                get_sql_diagnostics(HandleType::Dbc, conn_handle as Handle)
```

It looks like if SQLGetInfoW fails the error will be pushed onto the conn handle, so for the macro I think this might make sense (unless the env handle gets a diagnostic some other way).

Reply:
Thanks for catching that, I updated it to use the `conn_handle`